src/appleDeviceUtils/captureDdgGlobals.js

```diff
@@ -17,7 +17,30 @@ const secretGlobals = {
     JSONparse: window.JSON.parse,
     Arrayfrom: window.Array.from,
     Promise: window.Promise,
-    ObjectDefineProperty: window.Object.defineProperty
+    ObjectDefineProperty: window.Object.defineProperty,
+    capturedWebkitHandlers: {}
 }
 
-export default secretGlobals
+export { ddgGlobals }
+
+/**
+ * When required (such as on macos 10.x), capture the `postMessage` method on
+ * each webkit messageHandler
+ *
+ * @param {string[]} handlerNames
+ */
+export function captureWebkitHandlers (handlerNames) {
+    for (let webkitMessageHandlerName of handlerNames) {
+        if (typeof window.webkit.messageHandlers?.[webkitMessageHandlerName]?.postMessage === 'function') {
+            ddgGlobals.capturedWebkitHandlers[webkitMessageHandlerName] = window.webkit.messageHandlers[webkitMessageHandlerName].postMessage.bind(window.webkit.messageHandlers[webkitMessageHandlerName])
+            Object.defineProperty(window.webkit.messageHandlers[webkitMessageHandlerName], 'postMessage', {
+                writable: false,
+                configurable: false,
+                enumerable: false,
+                value () {
+                    /** no-op */
+                }
+            })
```

Comment:
Should we also remove these handlers from the scope? Like `webkit.messageHandlers[handler] = {}` or something like that?

Reply:
Those are not writable :)